src/compass/s1_geocode_metadata.py

```diff
@@ -236,16 +237,47 @@ def geocode_luts(geo_burst_h5, burst, cfg, dst_group_path, item_dict,
             isce3.io.Raster(
                 f"IH5:::ID={dst_dataset.id.id}".encode("utf-8"), update=True)
 
-        # populate and prepare radargrid LUT input raster
+        # Define the radargrid for LUT interpolation
+        # The resultant radargrid will have
+        # the very first and the last LUT values be included.
+        radargrid_interp = burst.as_isce3_radargrid()
+        radargrid_interp.width = int(np.ceil(burst.width / dec_factor[0]))
+        range_px_interp_vec = np.linspace(0, burst.width - 1, radargrid_interp.width)
+        intv_interp_range = range_px_interp_vec[1] - range_px_interp_vec[0]
+
+        radargrid_interp.length = int(np.ceil(burst.length / dec_factor[1]))
+        azimuth_px_interp_vec = np.linspace(0, burst.length - 1, radargrid_interp.length)
+        intv_interp_azimuth = azimuth_px_interp_vec[1] - azimuth_px_interp_vec[0]
+
+        radargrid_interp.range_pixel_spacing *= intv_interp_range
+        radargrid_interp.prf /= intv_interp_azimuth
```

Comment:
I think this could be useful elsewhere so let's move this block into function `decimate_radar_grid`  [`src/compass/utils/radar_grid.py`](https://github.com/opera-adt/COMPASS/blob/9a35bed3be8f2dce5ebc525ed45292d74a196761/src/compass/utils/radar_grid.py).

Also let's move this function outside the the `for` loop above. AFAICT the decimated and normal radar grid does not change over the iterated items.

Reply:
Good idea. The radargrid decimation routine is moved to a new function `get_decimated_rdr_grd()` in `radargrid.py`